Source/WebCore/css/CSSCounterStyleDescriptors.cpp

```diff
@@ -61,99 +62,112 @@ static CSSCounterStyleDescriptors::Ranges translateRangeFromStyleProperties(cons
 static String symbolToString(const CSSValue* value)
 {
     if (!value || !value->isPrimitiveValue())
-        return String();
+        return { };
 
     auto& primitiveValue = downcast<CSSPrimitiveValue>(*value);
     return primitiveValue.stringValue();
 }
 
+static CSSCounterStyleDescriptors::AdditiveSymbols translateAdditiveSymbolsFromStyleProperties(const StyleProperties& properties)
+{
+    auto value = properties.getPropertyCSSValue(CSSPropertyAdditiveSymbols);
+    if (!value)
+        return { };
+
+    auto& additiveSymbols = downcast<CSSValueList>(*value);
+    CSSCounterStyleDescriptors::AdditiveSymbols result;
+    for (auto& additiveSymbol : additiveSymbols) {
+        if (auto* pair = dynamicDowncast<CSSValuePair>(&additiveSymbol)) {
+            auto weight = downcast<CSSPrimitiveValue>(pair->first()).value<unsigned>();
+            auto symbol = symbolToString(&pair->second());
+            result.constructAndAppend(symbol, weight);
+        } else
+            return { };
```

Comment:
I think all the non pair cases are rejected by `consumeCounterStyleAdditiveSymbols`, so you could just `auto& pair = downcast<CSSValuePair>(additiveSymbol);` directly and replace `pair->` with `pair.`, and remove the `else` branch.